### PR TITLE
 Fix XSS in some dashboards queries

### PR DESCRIPTION
**Pull request by msaf1980**
Fixes XSS in some dashboards queries:
issue #2744
issue #2745

Also fix:
issue #2746

### CODE REVIEW

File: webapp/graphite/errors.py [hunk 1]
```diff
@@ -94,12 +95,22 @@ def __str__(self):
...
-    s = s.replace("<", "&lt;")
-    s = s.replace(">", "&gt;")
+def safe_param(name, s):
+    if is_unsafe_str(s):
+        raise InputParameterError("{} contain unsafe symbols".format(name))
+    return s
+
+
+def is_unclean_str(s):
+    for symbol in '&<>~!@#$%^*()`':
```
Comment (msaf1980) [inline webapp/graphite/errors.py:105]: Minimal set of unsafe symbols is  `&<>`. May be other symbols are accepted ?

Comment (deniszh) [reply webapp/graphite/errors.py:105]: Let's stick to wide set for now, we'll fix later, if needed

**Comment by risicle** [comment]
It would be nice to have a release incorporating this as it has now been indicated in 3 different CVEs: CVE-2022-4730, CVE-2022-4729 & CVE-2022-4728.

**Comment by deniszh** [comment]
Agreed, will do release soon

**Comment by deniszh** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|1.1.x|[](https://github.com/graphite-project/graphite-web/pull/2801)|


### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**